### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by MagdaPaj**
## Purpose

* Add documentation about edge-observability setup and Otel Collector configuration

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ This sample is currently work in progress.
 
 In this sample, a Visual Studio Code dev container is used as the developer sandbox to leverage a local K3D cluster, initialize Azure resources and install a baseline set of Azure IoT Operations' components.
 
+The objective of this repository is to illustrate how to setup observability to be able to locally vizualize health of the cluster and AIO components.
```
Comment (katriendg) [inline README.md]: AIO is not an approved acronym we can use so can you just use 'Azure IoT Operations' components fully written?

File: docs/images/otel-diagram.jpg
Comment (katriendg) [inline docs/images/otel-diagram.jpg:1]: Really nice clear overview!

File: docs/observability-setup.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+The local edge observability stack is composed of the following components:
+
+* Grafana for interactive visualizations and analytics,
+* Tempo for tracing,
+* Loki for logging,
+* Prometheus for metrics.
+
+These components are deployed to the cluster using a GitOps approach through Flux. The `app` folder contains [edge-observability](./apps/edge-observability/) directory where all components are configured. This is achieved by specifying a Helm repository and then configuring a Helm release for each component with the minimal required setup.
+
+The Prometheus Helm chart deploys the Prometheus server, the Grafana Loki Helm chart deploys Loki, and the Grafana Tempo Helm chart deploys Tempo. The Grafana Helm chart configures data sources to visualize data from Prometheus, Loki, and Tempo. It also deploys a set of dashboards. These include custom dashboards found in the [grafana-dashboards](./apps/edge-observability/1.0/grafana-dashboards/) folder and imported dashboards from grafana.com, such as the [OpenTelemetry Collector](https://grafana.com/dashboards/15983) and [Otel - kubeletstats](https://grafana.com/dashboards/18681) dashboards.
```
Comment (katriendg) [inline docs/observability-setup.md]: In the third sentence `custom dashboards` section, could we either detail further or add a reference to where the custom dashboards are coming from? For MQ, OPC UA, the dashboards are coming from the official repo from docs and we should point that out - https://learn.microsoft.com/en-us/azure/iot-operations/monitor/howto-configure-observability#deploy-dashboards-to-grafana

File: docs/observability-setup.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  - [Context](#context)
+  - [Local egde observability stack](#local-egde-observability-stack)
+  - [OpenTelemetry collector configuration](#opentelemetry-collector-configuration)
+  - [Grafana dashboards updates](#grafana-dashboards-updates)
+  - [Considerations](#considerations)
+  - [Troubleshooting](#troubleshooting)
+
+## Context
+
+Current Azure IoT Operations preview template deploys the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) which is responsible for collecting metrics, traces, and logs from various Azure IoT Edge components.
```
Comment (katriendg) [inline docs/observability-setup.md]: I think the last words in this sentence are a typo 'Azure IoT Edge'. We could update to:

`Current Azure IoT Operations preview template deploys the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) which is responsible for collecting metrics, traces, and logs from its various  components.`

File: README.md (hunk 1 above)
Comment (katriendg) [inline README.md]: It's probably useful to add extend the note about the existing Azure IoT Operations observability documentation and integration on the cloud (as in official docs, link to it). This repo is to focus on an edge based visualization and that's what we want to call out.

Second point on this: we should also add the link to your new MD file either here or lower on the README.me file so folks can browse to it.

File: docs/observability-setup.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+Additionally
+
+* [prometheus](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver) receiver was updated to scrape MQ metrics.
+* opentelemetry-collector-contrib image was updated to version [0.95.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.95.0).
+
+The following diagram illustrates configured components and the flow of observability data:
+
+![otel-diagram](./images/otel-diagram.jpg)
+
+The OpenTelemetry Collector can be further customized according to your requirements. For instance, you might choose to export data to your preferred observability backend or limit the exported data by applying filters through processors. An example of this could be:
```
Comment (katriendg) [inline docs/observability-setup.md:72]: Love this section!

**Review by katriendg — COMMENTED** [review]
This is a great write-up, adding a lot of clarity and showing users how things were configured, calling out things to consider, ...
I have a few minor comments and then it is all ready to merge.
Thanks!

File: docs/observability-setup.md (hunk 2 above)
Comment (MagdaPaj) [reply docs/observability-setup.md]: MQ, and OPC UA dashboards were not exactly a copy paste from the sample dashboards, as we needed to change some data sources names, or metrics names, and we just wanted to have one dashboard for MQ and one for OPC UA Broker. But we built them based on these provided dashboards, so added a comment about it. Let me know WDYT.

Comment (katriendg) [reply docs/observability-setup.md]: Thanks, that makes sense and with your additional comment it's all clear to whoever will use it. Perfect

File: README.md (hunk 1 above)
Comment (katriendg) [reply README.md]: Perfect

**Review by katriendg — APPROVED** [review]
Looking really good, thanks so much for your contributions @MagdaPaj. Approving.

**Pull request merged**